src/arrayMethodSort.js

```diff
@@ -4,8 +4,42 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function(compareFunction = stringComperison) {
     // write code here
+    let counter;
+
+    do {
+      counter = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const prevElem = this[i - 1];
+        const currentElem = this[i];
+
+        if (compareFunction(prevElem, currentElem) > 0) {
+          this[i] = prevElem;
+          this[i - 1] = currentElem;
+
+          counter++;
+        }
+      }
+    } while (counter > 0);
+
+    return this;
+  };
+
+  const stringComperison = (prev, current) => {
+    const prevEl = '' + prev;
+    const curEl = '' + current;
+
+    if (prevEl > curEl) {
+      return 1;
+    }
+
+    if (prevEl < curEl) {
+      return -1;
+    }
+
+    return 0;
   };
```

Comment:
create arrow functions before you use them

Reply:
but functions are instantly fully initialized, or it's a rule of good code?

Reply:
it's rule of good code and for better readability to create all functions at the top